server/src/socobouser/routes/SocoboUserProviders.route.ts

```diff
@@ -0,0 +1,36 @@
+import { Router } from "express";
+import { ModelValidationHandler, ValidationGroup } from "../../app/index";
+import { AuthValidationHandler } from "../../auth/index";
+import { SocoboUserProvidersHandler } from "../handlers/SocoboUserProviders.handler";
+import { SocoboUserProvider } from "../index";
+
+export class SocoboUserProvidersRoute {
+
+  constructor (
+    private _router: Router,
+    private _socoboUserProvidersHandler: SocoboUserProvidersHandler,
+    private _authValidationHandler: AuthValidationHandler,
+    private _modelValidationHandler: ModelValidationHandler
+  ) {}
+
+  public createRoutes (): Router {
+    this._router.get("/",
+      this._authValidationHandler.checkToken,
+      this._socoboUserProvidersHandler.getAll);
+
+    this._router.get("/:id",
+      this._authValidationHandler.checkToken,
+      this._socoboUserProvidersHandler.getById);
+
+    this._router.put("/:id",
+      this._authValidationHandler.checkToken,
+      this._modelValidationHandler.validate(SocoboUserProvider, [ValidationGroup.USER]),
+      this._socoboUserProvidersHandler.updateById);
+
+    this._router.delete("/:id",
```

Comment:
I think providers should also be constant during the runtime of the app or why do you want to be able to delete a provider?

delete and update should only be possible for admins